src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+                streams_list.append(StaticFluoStream(stream.name.value, stream.raw[0]))
+
+        self.streams_list = streams_list
+
+        if self._tab_data_model.main.currentFeature.value.correlation_data and (self._tab_data_model.main.currentFeature.value.status.value in self._tab_data_model.main.currentFeature.value.correlation_data):
+            # Similar logic can be used while enabling the 3DCT button i.e. during the check of relevant conditions.
+            self.correlation_target = self._tab_data_model.main.currentFeature.value.correlation_data[
+                self._tab_data_model.main.currentFeature.value.status.value]
+            # Maintain the order of loading. First check the streams and then load the targets. If no streams are
+            # present or the saved streams are not available, load all the relevant streams, and reset the fib and fm
+            # targets accordingly.
+
+            # Load the streams
+            self.group_streams()
+            self._add_stream_group()
+
+            # Load the targets
+            targets = []
+            if self.correlation_target.fm_fiducials:
+                targets.append(self.correlation_target.fm_fiducials)
+            if self.correlation_target.fm_pois:
+                targets.append(self.correlation_target.fm_pois)
+            if self.correlation_target.fib_fiducials and self.correlation_target.fib_stream:
+                targets.append(self.correlation_target.fib_fiducials)
+            # flatten the list of lists
+            targets = list(
+                itertools.chain.from_iterable([x] if not isinstance(x, list) else x for x in targets))
+            self._tab_data_model.main.targets.value = targets
+            if self.correlation_target.fib_surface_fiducial:
+                self._tab_data_model.fib_surface_point.value = self.correlation_target.fib_surface_fiducial
+        else:
+            # initialize the correlation target
+            self._tab_data_model.main.currentFeature.value.correlation_data = {}
+            self._tab_data_model.main.currentFeature.value.correlation_data[
+                self._tab_data_model.main.currentFeature.value.status.value] = FIBFMCorrelationData()
+            self.correlation_target = self._tab_data_model.main.currentFeature.value.correlation_data[
+                self._tab_data_model.main.currentFeature.value.status.value]
+            self.group_streams()
+            self._add_stream_group()
+
+        self._panel.fp_correlation_panel.Show(True)
+
+    @call_in_wx_main
+    def _on_fm_streams_visiblity(self, stream_projections: ListVA) -> None:
+        """
+        Update the stream groups when the user selects the FM stream panel from a different group. The
+        group of the selected stream panel will be visible and the other groups will be invisible
+        as well as collapsed.
+        """
+        if not self.stream_groups or len(stream_projections) <= 0:
```

Comment:
The pythonic way of checking a list is empty is to just write `not stream_projections`.